saleor/graphql/webhook/mutations.py

```diff
@@ -301,3 +311,95 @@ def perform_mutation(cls, _root, info, **data):
         manager = get_plugin_manager_promise(info.context).get()
         manager.event_delivery_retry(delivery)
         return EventDeliveryRetry(delivery=delivery)
+
+
+class WebhookDryRunInput(graphene.InputObjectType):
+    object_id = graphene.ID(
+        description="The ID of an object to serialize.", required=True
+    )
+    query = graphene.String(
+        description="The subscription query that defines the webhook event and its "
+        "payload.",
+        required=True,
+    )
+
+
+class WebhookDryRun(BaseMutation):
+    payload = JSONString(
+        description="JSON payload, that would be sent out to webhook's target URL."
+    )
+
+    class Arguments:
+        id = graphene.ID(description="The ID of the webhook.", required=True)
+        input = WebhookDryRunInput(
+            description="Fields required to perform dry run of a webhook event.",
+            required=True,
+        )
+
+    class Meta:
+        description = "Performs a dry run of a webhook event."
+        permissions = (AppPermission.MANAGE_APPS,)
+        error_type_class = WebhookDryRunError
+
+    @classmethod
+    def validate_input(cls, info, **data):
+        query = data.get("input", {}).get("query")
+        object_id = data.get("input", {}).get("object_id")
+        webhook_id = data.get("id")
+
+        event_type = get_event_type_from_subscription(query)
+        if not event_type:
+            raise_validation_error(
+                field="query",
+                message="Can't parse an event type from query.",
+                code=WebhookDryRunErrorCode.GRAPHQL_ERROR,
+            )
+
+        event = WEBHOOK_TYPES_MAP.get(event_type)
+        if not event:
+            event_name = event_type[0].upper() + to_camel_case(event_type)[1:]
+            raise_validation_error(
+                field="query",
+                message=f"Event type: {event_name} is not defined in graphql schema.",
+                code=WebhookDryRunErrorCode.GRAPHQL_ERROR,
+            )
+
+        related_models = [
+            field.type.get_model().__name__
+            for field in event._meta.fields.values()
+            if hasattr(field, "type") and hasattr(field.type, "get_model")
+        ]
```

Comment:
We need to rethink this, for example event `ProductVariantOutOfStock` requires the `stock` object (see how `subscribable_object` is passed for webhooks).